tests/fermi/test_fermionic.py

```diff
@@ -96,37 +97,268 @@ def test_compact(self, fw, str_rep):
 
     @pytest.mark.parametrize("fw, str_rep", tup_fw_str)
     def test_str(self, fw, str_rep):
+        """Test __str__ and __repr__ methods"""
         assert str(fw) == str_rep
         assert repr(fw) == str_rep
 
-    tup_fw_mult = (
-        (fw1, fw1, FermiWord({(0, 0): "+", (1, 1): "-", (2, 0): "+", (3, 1): "-"})),
+    def test_pickling(self):
+        """Check that FermiWords can be pickled and unpickled."""
+        fw = FermiWord({(0, 0): "+", (1, 1): "-"})
+        serialization = pickle.dumps(fw)
+        new_fw = pickle.loads(serialization)
+        assert fw == new_fw
+
+    @pytest.mark.parametrize(
+        "operator",
+        [
+            ({(0, 0): "+", (2, 1): "-"}),
+            ({(0, 0): "+", (1, 1): "-", (3, 0): "+", (4, 1): "-"}),
+            ({(-1, 0): "+", (0, 1): "-", (1, 0): "+", (2, 1): "-"}),
+        ],
+    )
+    def test_init_error(self, operator):
+        """Test that an error is raised if the operator orders are not correct."""
+        with pytest.raises(ValueError, match="The operator indices must belong to the set"):
+            FermiWord(operator)
```

Comment:
Why these tests are added in this PR?

Reply:
Okay, now I remember. These were moved (unchanged) from further down in the file, to group all of the tests concerning arithmetic operations and all of the other tests separately. So not added, but moved up from further down in the file :)